account_controllers/controllers.js

```diff
@@ -375,48 +390,70 @@ module.exports = {
 
         if (numSpotsToAdd > 0) {
           const newSpotIds = await Promise.all(
             Array.from({ length: numSpotsToAdd }, async () => {
-              const { error: newSpotError } = await supabase
-                .from('parking_spots')
-                .insert({
-                  garage_id: garageId,
-                })
-                .single();
-
-              if (newSpotError) {
-                console.error('Error adding new parking spot:', newSpotError);
+              try {
+                const { data: newSpotData, error: newSpotError } =
+                  await supabase
+                    .from('parking_spots')
+                    .insert({
+                      garage_id: garageId,
+                    })
+                    .select();
+
+                if (newSpotError) {
+                  console.error('Error adding new parking spot:', newSpotError);
+                  return null;
+                }
+
+                if (!newSpotData) {
+                  console.error(
+                    'Unexpected response from Supabase:',
+                    newSpotData
+                  );
+                  return null;
+                }
+
+                return newSpotData.id;
+              } catch (insertError) {
+                console.error('Error during insertion:', insertError);
                 return null;
               }
             })
           );
 
           const filteredNewSpotIds = newSpotIds.filter(
-            (spotsId) => spotsId !== null
+            (filteredid) => filteredid !== null
           );
-          const allSpotIds = existingSpotIds.concat(filteredNewSpotIds);
+          const allSpotIds = availableSpotIds.concat(filteredNewSpotIds);
+
+          await supabase.from('garages').update({ spots }).eq('id', garageId);
 
           res.status(200).json({ garageId, spotIds: allSpotIds });
         } else if (numSpotsToAdd < 0) {
-          const spotsToRemove = existingSpotIds.slice(0, -numSpotsToAdd);
-
-          const { error: removeSpotsError } = await supabase
-            .from('parking_spots')
-            .delete()
-            .in('id', spotsToRemove);
+          const spotsToRemove = availableSpotIds.slice(numSpotsToAdd);
 
-          if (removeSpotsError) {
-            console.error('Error removing parking spots:', removeSpotsError);
-            return res.sendStatus(500);
+          if (availableSpotIds.length < Math.abs(numSpotsToAdd)) {
+            return res.status(400).json({
+              error: 'Cannot remove spots due to conflicting reservations.',
+            });
           }
 
+          await supabase.from('parking_spots').delete().in('id', spotsToRemove);
+
+          await supabase.from('garages').update({ spots }).eq('id', garageId);
+
           res.status(200).json({ garageId, removedSpotIds: spotsToRemove });
         } else {
-          res.status(200).json({ garageId, spotIds: existingSpotIds });
```

Comment:
POSSIBLE EDGE CASE: nevermind, the case where the number of spots to add = 0. there might be an edge case in here.. what happens in the case where there are 10 total spots before the request, 5 spots with reservations, and the requested spots change is 5. => wouldnt that make the numSpotsToAdd=0 => and wouldnt that then mean this else block executes and we would have the number of number of spots updated in the garage table without the removal of any spots from the parking garage table? I thought when you walked me through examples on postman it handled this case correctly so if you check that case and it does fine then disregard this comment lol

Reply:
If there are 10 spots is change requested to 5 spots, but there are 5 spots of reservation with unique parking_spot_id, then 5 spots will be eliminated and the change request of 5 spots is executed. The total number of parking spots will be the 5 reservation with unique parking_spot_id.
numSpotsToAdd = 10 - 5
numSpotsToAdd = spots - existingSpotIds.length;
thus numSpotsToAdd  is 5

One thing to note is numSpotsToAdd is either + or - or 0. It doesn't have to be positive, only in cases where we want to add cars. Thus, in this case it never reaches to else statement because numSpotToAdd is not 0. Example you mentioned highlighted here in result:

10 total spots
![10spots](https://github.com/pokemon-parKing/parKing-account-server/assets/144174704/faa529be-191e-497d-b072-0598cc2002c7)


5 spots requested (means that we want the spots to change to 5) /w 5 unique parking spots reserved currently
![5spots](https://github.com/pokemon-parKing/parKing-account-server/assets/144174704/65262ce8-ff27-4cf1-afb5-5395fd5f9938)

Garage table spots updated per this request 
![garagestable5](https://github.com/pokemon-parKing/parKing-account-server/assets/144174704/c5d2074b-f8a7-49b0-9fbc-344ed67caeca)

Reply:
But to your first point on whether that else block needs to execute anything, I thought about this a bit and yea it probably doesn't need to execute anything, I could have that block send a status 200 and 'spots are the same as current spot or something'